pkg/scheduler/store.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"sync"
+)
+
+/*
+The relationship between relevant API objects (arrows indicate selections/references):
+ep -> loc -> st
+ep: edgeplacement; loc: location; st: synctarget
+
+To facilitate incremental updates, we need to maintain internal data structures:
+
+	ep <- loc <- st
+
+or in other words,
+
+	ep <- loc
+	loc <- st
+
+*/
+
+type empty struct{}
+
+type internalData struct {
+	l sync.Mutex
+
+	epsBySelectedLoc map[string]map[string]empty // ep <- loc
```

Comment:
If the inner map held the `Key` of https://github.com/kcp-dev/apimachinery/pull/70 (or an anticipatory local copy of that), or the `ExternalName` of pkg/placement, rather than `string` then you could eliminate a lot of code handling parsing errors.